modules/light-clients/06-solomachine/types/update_test.go

```diff
@@ -563,3 +563,56 @@ func (suite *SoloMachineTestSuite) TestVerifyClientMessageMisbehaviour() {
 		}
 	}
 }
+
+func (suite *SoloMachineTestSuite) TestUpdateState() {
+	var (
+		clientState exported.ClientState
+		header      exported.Header // TODO: Update to ClientMessage interface
+	)
+
+	// test singlesig and multisig public keys
+	for _, solomachine := range []*ibctesting.Solomachine{suite.solomachine, suite.solomachineMulti} {
+
+		testCases := []struct {
+			name    string
+			setup   func()
+			expPass bool
+		}{
+			{
+				"successful update",
+				func() {
+					clientState = solomachine.ClientState()
+					header = solomachine.CreateHeader()
+				},
+				true,
+			},
+		}
+
+		for _, tc := range testCases {
+			tc := tc
+
+			suite.Run(tc.name, func() {
+				// setup test
+				tc.setup()
+
+				clientState, ok := clientState.(*types.ClientState)
+				if ok {
+					cs, consensusState, err := clientState.UpdateState(suite.chainA.GetContext(), suite.chainA.Codec, suite.store, header)
```

Comment:
Assuming we will be removing the return values we can simply retrieve the client via `suite.store` in `tc.expPass` and then make assertions on that.